src/main/java/org/kohsuke/groovy/sandbox/SandboxTransformer.java

```diff
@@ -704,7 +704,7 @@ public void visitExpressionStatement(ExpressionStatement es) {
                     if (mightBePositionalArgumentConstructor((VariableExpression) leftExpression)) {
                         CastExpression ce = new CastExpression(leftExpression.getType(), de.getRightExpression());
                         ce.setCoerce(true);
-                        es.setExpression(transform(new DeclarationExpression(leftExpression, de.getOperation(), transform(ce))));
+                        es.setExpression(transform(new DeclarationExpression(leftExpression, de.getOperation(), ce)));
```

Comment:
@jglick - so there are existing tests in [`script-security`](https://github.com/jenkinsci/script-security-plugin/blob/ea5ce369c4ed064c64be9ac93a03c12e08d5d167/src/test/java/org/jenkinsci/plugins/scriptsecurity/sandbox/groovy/SandboxInterceptorTest.java#L598) and [`workflow-cps`](https://github.com/jenkinsci/workflow-cps-plugin/blob/9289c8f9add190cfce82ea7d888575f9eebe55ff/src/test/java/org/jenkinsci/plugins/workflow/cps/CpsFlowDefinition2Test.java#L200) for SECURITY-580 - do we need more than that?